Automattic/jetpack
# Jetpack settings page: Removing defaultProps from SearchableModules component

Fixes https://github.com/Automattic/jetpack/issues/38539

## Proposed changes:

* This PR removes the `defaultProps` and `propTypes` from the `SearchableModules` component, adding defaults to the one property when defining that property.
* This prevents a console warning - `Warning: Connect(SettingsForm): Support for defaultProps will be removed from memo components in a future major release. Use JavaScript default parameters instead` - due to the legacy usage of `defaultProps`.
* Ultimately the issue was that the `SearchableModules` component was a function component, despite wrapping a class, so the `defaultProps` usage was missed on an earlier sweep of function component `defaultProps` usage.


### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/38539

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

To replicate the issue:
* First make sure you have script debug enabled (`define ( 'SCRIPT_DEBUG, 'true');` in `wp-config.php`)
* Then, testing locally on trunk or using a Jurassic Ninja site and the Beta tester plugin using bleeding edge Jetpack, visit the settings page - `/wp-admin/admin.php?page=jetpack#/settings`.
* You should see the above console error.

To test the fix:
* Apply this PR, and visit the settings page again. The console warning should be gone. The search function should work as it did before.
